### PR TITLE
add Router to project

### PR DESCRIPTION
**Pull request by haotran-ptc**

### CODE REVIEW

File: boilerplate-swiftui-bloc/boilerplate_swiftui_blocApp.swift [hunk 1]
```diff
@@ -11,11 +11,18 @@ import SwiftUI
...
+                )
             }
+            .environment(\.router, .main)
+            .navigationViewStyle(StackNavigationViewStyle())
         }
     }
 }
+
+extension NavigationRouter {
+  static var main = NavigationRouter(routes: .all)
```
Comment (camapblue) [inline boilerplate-swiftui-bloc/boilerplate_swiftui_blocApp.swift]: Why we keep static variable here ?

File: boilerplate-swiftui-bloc/Routing/Routes.swift [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            "bloc": contactBloc,
+        ])
+    }
+}
+
+extension Array where Element == NavigationRoute {
+    static var all: [NavigationRoute] {
+        let splash = NavigationRoute(path: "/splash", destination: SplashView())
+        let storyBook = NavigationRoute(path: "/storyBook", destination: Storybook())
+        let contactList = NavigationRoute(path: "/contactList", destination: ContactListView())
```
Comment (camapblue) [inline boilerplate-swiftui-bloc/Routing/Routes.swift]: Can we handle BlocProvider for ContactListView here

File: boilerplate-swiftui-bloc/boilerplate_swiftui_blocApp.swift (hunk 1 above)
Comment (haotran-ptc) [reply boilerplate-swiftui-bloc/boilerplate_swiftui_blocApp.swift]: I have updated this static variable to create within the App!

File: boilerplate-swiftui-bloc/Routing/Routes.swift (hunk 2 above)
Comment (haotran-ptc) [reply boilerplate-swiftui-bloc/Routing/Routes.swift]: Yes we can, I've updated the code!

**Review by camapblue — APPROVED** [review]

**Pull request merged**